metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -356,7 +367,9 @@ public void replay(PartitionChangeRecord record) {
         brokersToIsrs.update(record.topicId(), record.partitionId(),
             prevPartitionInfo.isr, newPartitionInfo.isr, prevPartitionInfo.leader,
             newPartitionInfo.leader);
-        log.debug("Applied ISR change record: {}", record.toString());
+        String topicPart = topicInfo.name + "-" + record.partitionId() + " with topic ID " +
```

Comment:
In the old controller, we bump up the leader epoch if the ISR is changed during the controlled shutdown. This helps prevent the shutting down broker from being added to ISR again. In the raft controller, we bump up the partitionEpoch when the ISR is changed. Do we plan to fence a fetch request with unmatched partitionEpoch to achieve the same logic? If so, do we have a jira to track that?

Reply:
Filed https://issues.apache.org/jira/browse/KAFKA-12733 for this